nautobot/dcim/tests/test_api.py

```diff
@@ -2410,3 +2443,130 @@ def setUpTestData(cls):
             "failover_strategy": "active-passive",
             "status": statuses[1].pk,
         }
+        # FIXME(jathan): The writable serializer for `status` takes the
+        # status `name` (str) and not the `pk` (int). Do not validate this
+        # field right now, since we are asserting that it does create correctly.
+        #
+        # The test code for `nautobot.core.testing.views.TestCase.model_to_dict()`
+        # needs to be enhanced to use the actual API serializers when `api=True`
+        cls.validation_excluded_fields = ["status"]
+
+
+class InterfaceRedundancyGroupTestCase(APIViewTestCases.APIViewTestCase):
+    model = InterfaceRedundancyGroup
+    brief_fields = ["display", "id", "name", "protocol", "url"]
+    choices_fields = ["protocol"]
+    # FIXME(jathan): The writable serializer for `status` takes the
+    # status `name` (str) and not the `pk` (int). Do not validate this
+    # field right now, since we are asserting that it does create correctly.
+    #
+    # The test code for `nautobot.core.testing.views.TestCase.model_to_dict()`
+    # needs to be enhanced to use the actual API serializers when `api=True`
+    validation_excluded_fields = ["status"]
+
+    @classmethod
+    def setUpTestData(cls):
+        statuses = Status.objects.get_for_model(InterfaceRedundancyGroup)
+        ips = IPAddress.objects.all()
+        secrets_groups = (
+            SecretsGroup.objects.create(name="Secrets Group 1"),
+            SecretsGroup.objects.create(name="Secrets Group 2"),
+            SecretsGroup.objects.create(name="Secrets Group 3"),
+        )
+        # Populating the data secrets_group and virtual_ip here.
+        cls.create_data = [
+            {
+                "name": "Interface Redundancy Group 4",
+                "protocol": "hsrp",
+                "status": "Active",
```

Comment:
This surprised me, because I thought it would fail! Tests are passing here because of #4188 - even though `status` is in the 2.0 REST API represented as a related object rather than a pseudo-enum, this is still/again permissible because `status.name` is the composite-key for Status records, and the REST API now allows for use of composite-keys as an alternative to PKs.